src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -315,6 +320,11 @@ private static InputStreamWithMetadata compressStreamWithGZIP(
 
       countingStream.flush();
 
+      // Normal flow will never hit here. This is only for testing purposes
```

Comment:
Can you write a utility method as same stuff is there multiple time.

Reply:
I added a method to check if the exception injection was enabled. The other parts are checking specific exceptions that the method could throw. Extracting this to a utility method may result in having to change the signatures of the methods to handle additional exceptions they wouldn't throw in normal flow.